custom_components/hafas/sensor.py

```diff
@@ -124,17 +128,28 @@ async def async_update(self) -> None:
 
         products = {product: (product in self.products) for product in self.client.profile.availableProducts}
         try:
-            self.journeys = await self.hass.async_add_executor_job(
-                functools.partial(
-                    self.client.journeys,
-                    origin=self.origin,
-                    destination=self.destination,
-                    date=dt_util.as_local(dt_util.utcnow() + self.offset),
-                    max_changes=0 if self.only_direct else -1,
-                    max_journeys=3,
-                    products=products
+            if self.destination:
+                self.journeys = await self.hass.async_add_executor_job(
+                    functools.partial(
+                        self.client.journeys,
+                        origin=self.origin,
+                        destination=self.destination,
+                        date=dt_util.as_local(dt_util.utcnow() + self.offset),
+                        max_changes=0 if self.only_direct else -1,
+                        max_journeys=3,
```

Comment:
Should we change this to 10, too? Why would we have different max_journey?

Reply:
I don't recall why I put 3 there. I'll fix it.